File: hang.liu/905.js
var sortArrayByParity = function(A) {
    var left = 0
    var right = A.length - 1
    
    while (left < right) {
        while(A[left] % 2 !== 1 && left < right) {
            left++
        }
        
        while(A[right] % 2 !== 0 && left < right) {
            right--
        }
        
        var temp = A[left]
        A[left] = A[right]
        A[right] = temp
    }
    
    return A
    
};

File: xiang.li/905.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int i = 0;
        int j = A.length - 1;
        int temp;
        while(i < j){
            if(A[i] % 2 != 0) {
                temp = A[i];
                A[i] = A[j];
                A[j] = temp;
                j--;
            } else {
                i++;
            }
        }
        return A;
    }
}
